### PR TITLE
Add ADR about message visibility timeouts for file checks

### PR DESCRIPTION
**Pull request by suzannehamilton**

### CODE REVIEW

File: architecture-decision-records/0020-sqs-visibility-timeout.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+[Lambda concurrency limits]: https://zaccharles.medium.com/lambda-concurrency-limits-and-sqs-triggers-dont-mix-well-sometimes-eb23d90122e0
+[sqs-lambda]: https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html#events-sqs-queueconfig
+
+## Decision
+
+Set the visibility timeout to 3 times the Lambda timeout for all file check
+Lambdas (including the file download and API update Lambdas). From some initial
+testing, this is enough to fix the invalid failure messages.
+
+This change will mean that retries take much longer than the do at the moment.
```
Comment (TomJKing) [inline architecture-decision-records/0020-sqs-visibility-timeout.md:137]: Spelling: " ... than **_they_** do at ..."

**Review by TomJKing — COMMENTED** [review]
Looks good. Just a couple of editorial suggestions. Thanks for working through this and writing it up.

Might be worth @MancunianSam having a read through as well.

File: architecture-decision-records/0020-sqs-visibility-timeout.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+All of the file check SQS queues currently have `maxReceiveCount` set to 3,
+which means each message should be processed by Lambda up to 3 times until it
+succeeds. If every attempt fails, the queues are configured to send the message
+to a dead letter queue called `tdr-backend-check-failure-<env>`.
+
+This was mostly working as intended, but we noticed that some messages were
+occasionally sent to the failure queue even when the Lambda eventually succeded.
+This is a problem because it makes it hard to distinguish between file checks
+that completely failed and file checks that succeeded when retried. You could
+work out the answer for a specific file by checking the file check logs, but the
```
[the remark this answers is not in the document]
Comment (suzannehamilton) [reply architecture-decision-records/0020-sqs-visibility-timeout.md]: Fixed, thanks.

File: architecture-decision-records/0020-sqs-visibility-timeout.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+messages are processed, the number of message on the _failure_ queue grows from
+135 to 241. It looks like 106 file check messages failed to be processed at all.
+
+The state of the failure queue isn't reflected in the database, though. All of
+the file check results have been stored, and the user is taken to the file check
+results page as normal.
+
+### Cause of bug
+
+We're not completely sure this is the cause, but as far as well can tell, the
```
[the remark this answers is not in the document]
Comment (suzannehamilton) [reply architecture-decision-records/0020-sqs-visibility-timeout.md]: Fixed, thanks.

File: architecture-decision-records/0020-sqs-visibility-timeout.md (hunk 1 above)
Comment (suzannehamilton) [reply architecture-decision-records/0020-sqs-visibility-timeout.md]: Fixed, thanks.

**Review by TomJKing — APPROVED** [review]
Looks good.

**Pull request merged**